ReactNativeClient/lib/models/Setting.js

```diff
@@ -918,7 +940,11 @@ class Setting extends BaseModel {
 		for (let i = 0; i < this.cache_.length; i++) {
 			const s = Object.assign({}, this.cache_[i]);
 			s.value = this.valueToString(s.key, s.value);
-			queries.push(Database.insertQuery(this.tableName(), s));
+			if (this.constants_.appType !== 'mobile' && this.isSecureKey(s.key)) {
```

Comment:
Again no device check in Setting.js. Use the shim

Reply:
Sure!